ARCs/arc-0060.md

```diff
@@ -0,0 +1,506 @@
+
+#### Interface `Ed25519Pk`
+
+An `Ed25519Pk` object is a 32-byte public key, point of the ed25519 elliptic curve. It is base64 encoded in accordance with <a href="https://www.ietf.org/rfc/rfc7468.txt">RFC-7468</a>. The key is not transformed into an Algorand address.
+
+```tsx
+export type Ed25519Pk = string;
+```
+
+> The wallet **MAY** want to operate with standard Algorand address directly. The transformation from a generic key to an Algorand address is left to the implementation. See <a href="https://developer.algorand.org/docs/get-details/accounts/#transformation-public-key-to-algorand-address">Public Key to Algorand Address</a> section of the developer portal.
+
+#### Interface `SignedDataStr`
+
+`SignedDataStr` is the produced 64-byte array ed25519 digital signature of the `StdData` object.
+
+```tsx
+export type SignedDataStr = string;
+```
+
+#### Interface `SignedData`
+
+A `SignedData` object is a list of `SignedDataStr` representing the signed data. The list must contain all the signatures in case of a multisig.
+
+```tsx
+export type SignedData = SignedDataStr[];
+```
+
+#### Interface `StdSignData`
+
+A `StdSignData` object represents a structured byte array of data to be signed by a wallet.
+
+```tsx
+export interface StdSignData {
+    /**
+    * The structured arbitrary data to be signed.
+    */
+    data: StdData;
+    
+    /**
+    * A non-empty list of ed25519 public keys that must sign the data.
+    */
+    signers: Ed25519Pk[];
+
+    /**
+    * Optional delegated public key used to sign data on behalf of a signer.
+    * It can be used to sign data with rekeyed Algorand accounts.
+    */
+    authAddr?: Ed25519Pk;
+
+    /**
+    * Optional multisig metadata used to sign the data.
+    */
+    msig?: MultisigMetadata;
+
+    /**
+    * Optional HD wallet metadata used to derive
+    * the public key used to sign the data.
+    */
+    hdPath?: HDWalletMetadata
+}
+```
+
+The `msig` object is defined in <a href="https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0001.md">ARC-1</a> and it specifies the metadata of an Algorand multisig address.
+
+This ARC assumes the `msig.addrs` to be a list of `Ed25519Pk` non transformed into an Algorand Address.
+
+> From <a href="https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0001.md">ARC-1</a> an `AlgorandAddress` is a transformed ed25519 public key represented by a 58-character base32 string that includes a checksum.
+
+#### Enum `ScopeType`
+
+The `ScopeType` enumerates constant strings with the scope of a signing action.
+
+This ARC introduces only one scope for signing arbitrary bytes.
+
+| ScopeType | Description |
+| --- | --- |
+| ARBITRARY | Signature of an array of arbitrary bytes. This is the most generic scope. |
```

Comment:
Would it make sense to add the Scopes now? i.e `Authorization`, `LSig`, etc ? 

This way, depending on the Scope the wallet knows which schema to validate against

Reply:
I think it makes sens for `Lsig`. For Auth, I would introduce the new `ScopeType` with ARC-31.

Reply:
Resolving since we have `AUTH` and `LSIG` scopes (+ schemas) now.